packages/NodeTypeResolver/PHPStan/Scope/PHPStanNodeScopeResolver.php

```diff
@@ -149,13 +144,17 @@ private function processNodesWithMixinHandling(
      */
     private function isMixinInSource(array $nodes, SmartFileInfo $smartFileInfo): bool
     {
-        $realPath = $smartFileInfo->getRealPath();
-        return (bool) $this->betterNodeFinder->findFirst($nodes, function (Node $node) use ($realPath): bool {
-            if (! $node instanceof FullyQualified) {
+        return (bool) $this->betterNodeFinder->findFirst($nodes, function (Node $node): bool {
+
+            if (! $node instanceof FullyQualified && ! $node instanceof Class_) {
                 return false;
             }
 
-            $className = $node->toString();
+            if ($node instanceof Class_ && $node->isAnonymous()) {
```

Comment:
you should can use ClassAnalyzer service for detecting anonymous class

Reply:
I tried to use ClassAnalyzer service for detecting anonymous class, but it cannot ensure that the property `namespacedName` exists.

Reply:
it possibly needs a node instanceof Class_ as well, like this:

```
$node instanceof Class_ && $this->classAnalyzer->isAnonymous()
```

if that still not works, possibly revert is the way for it as it possibly overlapped with filling namespacedName property

Reply:
`$this->classAnalyzer->isAnonymous()` will return false like this:
```php
new class extends ParentClass implements SomeInterface
{
    use AnotherTrait;
};
```
but the property namespacedName not exists in this case.

Temporarily reverted to make the test pass.